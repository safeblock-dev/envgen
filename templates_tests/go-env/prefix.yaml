options:
  go_package: prefix

groups:
  - name: App
    description: Application settings
    prefix: APP_
    fields:
      - name: debug
        type: bool
        description: Enable debug mode
        default: "false"
      - name: port
        type: int
        description: Server port
        required: true
        example: "8080"
  - name: Database
    description: Database settings
    prefix: DB_
    fields:
      - name: host
        type: string
        description: Database host
        default: "localhost"
      - name: port
        type: int
        description: Database port
        default: "5432" 