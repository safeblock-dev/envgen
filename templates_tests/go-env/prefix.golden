// Code generated by envgen. DO NOT EDIT.
// This file was automatically generated and should not be modified manually.

//go:generate go run github.com/safeblock-dev/envgen/cmd/envgen -c prefix.yaml -o prefix.generated -t ../../templates/go-env

package prefix

// AppConfig represents Application settings
type AppConfig struct {
	Debug bool `env:"APP_DEBUG" envDefault:"false"` // Enable debug mode
	Port int `env:"APP_PORT,required"` // Server port
}

// DatabaseConfig represents Database settings
type DatabaseConfig struct {
	Host string `env:"DB_HOST" envDefault:"localhost"` // Database host
	Port int `env:"DB_PORT" envDefault:"5432"` // Database port
}